builder/dockerfile/copy.go

```diff
@@ -68,10 +68,10 @@ type copier struct {
 	tmpPaths    []string
 }
 
-func copierFromDispatchRequest(req dispatchRequest, download sourceDownloader, imageSource *imageMount) copier {
+func newCopier(source builder.Source, cache pathCache, download sourceDownloader, imageSource *imageMount) copier {
```

Comment:
Seems to me that `newCopier` is still called from `dispatchRequest`. Just these properties are extracted on caller side.